api/src/wfl/module/covid.clj

```diff
@@ -17,6 +19,57 @@
 (defn ^:private get-snapshots-from-workspace
   [workspace])
 
+(defn throw-unless-column-exists [dataset table column-name]
+  "Throw or return the column from `table` with `column-name`"
+  (let [[result & more :as all] (-> table
+                                    (get-in [:columns])
+                                    (->> (filter (comp #{column-name} :name))))]
+    (when-not result
+      (throw (ex-info "No column with name" {:name column-name :dataset dataset})))
+    result))
+
+(defn throw-unless-table-exists [dataset table-name column-name]
+  "Throw or return the table from `dataset` with `table-name`"
+  (let [[result & more :as all] (-> dataset
+                                    (get-in [:schema :tables])
+                                    (->> (filter (comp #{table-name} :name))))]
+    (when-not result
+      (throw (ex-info "No table with name" {:name table-name :dataset dataset})))
+    (when result
+      (throw-unless-column-exists dataset result column-name))
+    result))
+
+(defn verify-source!
+  "Verify that the `dataset` exists and that the WFL has the necessary permissions to read it"
+  [{:keys [name dataset] :as source}]
+  (when-not (= (:name source) "Terra DataRepo")
+    (throw (ex-info "Unknown Source" {:source source})))
+  (try
+    (-> (datarepo/dataset (:dataset source))
+        (throw-unless-table-exists "flowcell" "updated"))
+    (catch Throwable t
+      (throw (ex-info "Cannot access Dataset" {:dataset dataset
+                                               :cause   (.getMessage t)})))))
+
+(defn verify-executor!
+  "Verify the method-configuration exists."
+  [{:keys [name method_configuration] :as executor}]
+  (when-not (= (:name executor) "Terra")
+    (throw (ex-info "Unknown Executor" {:executor executor})))
+  (when-not (:method_configuration executor)
+    (throw (ex-info "Unknown Method Configuration" {:executor executor}))))
+
+(defn verify-sink!
+  "Verify that the WFL has access to both firecloud and the `workspace`."
+  [{:keys [name workspace] :as sink}]
+  (when-not (= (:name sink) "Terra Workspace")
+    (throw (ex-info "Unknown Sink" {:sink sink})))
+  (try
+    (firecloud/get-workspace workspace)
+    (catch Throwable t
+      (throw (ex-info "Cannot access the workspace" {:workspace workspace
+                                                     :cause     (.getMessage t)})))))
```

Comment:
Suggest making `firecloud/get-workspace-json` public and calling it here.
Better yet, rename `get-workspace-json` to `get-workspace` and so on.

Reply:
Done :)